src/components/dropdown/dropdown.svelte

```diff
@@ -0,0 +1,244 @@
+<script lang="ts">
+  import { createEventDispatcher } from 'svelte'
+  import clickOutside from '../../directives/clickOutside'
+  import Control from '../control/control.svelte'
+  import { scale } from 'svelte/transition'
+  import Icon from '../icon/icon.svelte'
+
+  export let placeholder = ''
+  export let value: string = ''
+  export let disabled = false
+
+  let dispatch = createEventDispatcher()
+
+  let isOpen = false
+  let popup: HTMLDivElement
+  let button: HTMLButtonElement
+
+  // Work out what options have been slotted into the control. We need to handle
+  // being in a web-component (via assignedElements) and Svelte's internal
+  // slotting separately.
+  $: options = Array.from(
+    (
+      popup?.querySelector('.leo-dropdown-popup slot') as HTMLSlotElement
+    )?.assignedElements() ??
+      popup?.querySelectorAll('.leo-dropdown-popup > *') ??
+      []
+  )
+
+  // When the options change, we should assign each one a tabindex - this let's
+  // us handle changing the selection with the keyboard (tab or arrow keys).
+  $: {
+    for (const [option, index] of options.map((o, i) => [o, i] as const)) {
+      option.setAttribute('tabindex', (index + 1).toString())
+    }
+  }
+
+  /**
+   * Selects an option from an element
+   * @param option The element containing the option.
+   */
+  function selectOption(e: Event) {
+    // Find the option which was clicked on, if any.
+    const option = options.find((option) => e.composedPath().includes(option))
+
+    // If the event was triggered for something which isn't an option don't fire
+    // a change event.
+    if (!option) return
+
+    // If the option element doesn't have a value, fallback to using the text
+    // content - this allows writing simplified options:
+    // i.e. <o>1</o>
+    value = option.getAttribute('value') ?? option.textContent
```

Comment:
I wasn't sure if I should do this - maybe we don't want to fall back to the text content of the node? We could enforce that all children have the `value` attribute?

Reply:
I suppose it seems fine, but most of the time I'd imagine we'd have translated text.

Reply:
Yeah - maybe it just doesn't buy us anything? Pretty happy to just remove it, if you prefer?